test/groovy/BuildExecuteTest.groovy

```diff
@@ -315,6 +360,49 @@ class BuildExecuteTest extends BasePiperTest {
         assertThat(buildToolCalled, is(true))
     }
 
+    @Test
+    void testCnbBuildCalledWhenConfigured() {
+        def cnbBuildCalled = false
+        def npmExecuteScriptsCalled = false
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], { m ->
+            npmExecuteScriptsCalled = true
+        })
+        helper.registerAllowedMethod('cnbBuild', [Map.class], { m ->
+            cnbBuildCalled = true
+            return
+        })
+        assertThat(nullScript.commonPipelineEnvironment.getContainerProperty('buildpacks'), nullValue())
+
+        stepRule.step.buildExecute(
+            script: nullScript,
+            buildTool: 'npm',
+            cnbBuild: true
+        )
+
+        assertThat(npmExecuteScriptsCalled, is(true))
+        assertThat(cnbBuildCalled, is(true))
+    }
+
+    @Test
+    void testCnbBuildNotCalledWhenNotConfigured() {
+        def cnbBuildCalled = false
+        def npmExecuteScriptsCalled = false
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], { m ->
+            npmExecuteScriptsCalled = true
+        })
+        helper.registerAllowedMethod('cnbBuild', [Map.class], { m ->
+            cnbBuildCalled = true
+            return
+        })
+        stepRule.step.buildExecute(
+            script: nullScript,
+            buildTool: 'npm',
+            cnbBuild: false
+        )
+        assertThat(npmExecuteScriptsCalled, is(true))
+        assertThat(cnbBuildCalled, is(false))
+    }
+
```

Comment:
Add "testCnbBuildWithoutPrecedingBuild". Are we supporting this? How?

Reply:
No, we're not supporting this